app/src/main/java/com/example/parkingsystemjava/mvp/view/ParkingReservationView.java

```diff
@@ -2,22 +2,76 @@
 
 import android.app.Activity;
 import android.app.FragmentManager;
+import android.content.Context;
+import android.widget.Toast;
+import com.example.parkingsystemjava.R;
+import com.example.parkingsystemjava.databinding.ActivityReservationBinding;
 import com.example.parkingsystemjava.fragments.DateReservationDialogFragment;
+import com.example.parkingsystemjava.listener.ListenerDateTime;
 import com.example.parkingsystemjava.mvp.contract.ParkingReservationContract;
 import com.example.parkingsystemjava.mvp.view.base.ActivityView;
+import com.example.parkingsystemjava.utils.Constants;
+import com.example.parkingsystemjava.utils.ReservationErrorCodes;
 
-public class ParkingReservationView extends ActivityView implements ParkingReservationContract.ReservationActivityView {
+public class ParkingReservationView extends ActivityView implements ParkingReservationContract.ParkingReservationView {
 
-    private static final String DIALOG_FRAGMENT_ENTRY_DATE_PICKER = "DIALOG_FRAGMENT_ENTRY_DATE_PICKER";
+    private final ActivityReservationBinding binding;
 
-    public ParkingReservationView(Activity activity) {
+    public ParkingReservationView(Activity activity, ActivityReservationBinding binding) {
         super(activity);
+        this.binding = binding;
     }
 
     @Override
-    public void showEntryDatePicker() {
-        DateReservationDialogFragment entryReservationDialogFragment = DateReservationDialogFragment.newInstance();
+    public void showDatePicker(ListenerDateTime listenerDateTime, boolean dateSelector) {
+        DateReservationDialogFragment entryReservationDialogFragment = DateReservationDialogFragment.newInstance(listenerDateTime, dateSelector);
         FragmentManager fragmentManager = getFragmentManager();
-        entryReservationDialogFragment.show(fragmentManager, DIALOG_FRAGMENT_ENTRY_DATE_PICKER);
+        entryReservationDialogFragment.show(fragmentManager, Constants.DIALOG_FRAGMENT_ENTRY_DATE_PICKER);
+    }
+
+    @Override
+    public void showEntryDateSelected(String date) {
+        binding.editTextReservationActivityEntry.setText(date);
+    }
+
+    @Override
+    public void showExitDateSelected(String date) {
+        binding.editTextReservationActivityExit.setText(date);
+    }
+
+    @Override
+    public void showConfirmationMessage(ReservationErrorCodes errorCode) {
+        Context context = getContext();
+        if (context != null) {
+            Toast.makeText(context, getErrorMessaje(errorCode), Toast.LENGTH_LONG).show();
+        }
+    }
+
+    private String getErrorMessaje(ReservationErrorCodes errorCode) {
```

Comment:
this is not fine, remember that the view should not have logic.. this should be in the presenter and from the presenter call a specific method for each error

Reply:
done!